apps/dapp/frontend/components/savings/SavingsGoalCard.tsx

```
@@ -14,14 +59,14 @@ export function SavingsGoalCard({
   onDeposit: (g: SavingsGoal) => void;
   onArchive: (id: string) => void;
 }) {
-  const current = typeof goal.current_amount === "number" ? goal.current_amount : parseFloat(String(goal.current_amount)) || 0;
-  const target = typeof goal.target_amount === "number" ? goal.target_amount : parseFloat(String(goal.target_amount)) || 0;
-  const progress = Math.min(100, Math.max(0, goal.progress_pct ?? (target > 0 ? (current / target) * 100 : 0)));
-
+  const current = toNumber(goal.current_amount);
+  const target = toNumber(goal.target_amount);
   const deadlineDate = goal.deadline ? new Date(goal.deadline) : null;
   const daysLeft = deadlineDate ? (isBefore(deadlineDate, new Date()) ? 0 : Math.max(0, Math.ceil((deadlineDate.getTime() - Date.now()) / 86400000))) : null;
```

ESLint

[error] Error: Cannot call impure function during render

`Date.now` is an impure function. Calling an impure function can produce unstable results that update unpredictably when the component happens to re-render. (https://react.dev/reference/rules/components-and-hooks-must-be-pure#components-and-hooks-must-be-idempotent).

/home/jailuser/git/apps/dapp/frontend/components/savings/SavingsGoalCard.tsx:65:125
  63 |   const target = toNumber(goal.target_amount);
  64 |   const deadlineDate = goal.deadline ? new Date(goal.deadline) : null;
> 65 |   const daysLeft = deadlineDate ? (isBefore(deadlineDate, new Date()) ? 0 : Math.max(0, Math.ceil((deadlineDate.getTime() - Date.now()) / 86400000))) : null;
     |                                                                                                                             ^^^^^^^^^^ Cannot call impure function
  66 |   const deadlineLabel = deadlineDate ? (daysLeft === 0 ? "Due today" : `${daysLeft} day${(daysLeft ?? 0) > 1 ? "s" : ""} left`) : "—";
  67 |
  68 |   const hasRichData = goal.principal_amount != null || goal.locked_positions?.length;

(react-hooks/purity)